ext/reflection/php_reflection.c

```diff
@@ -4128,16 +4128,21 @@ ZEND_METHOD(ReflectionClass, getStaticPropertyValue)
 	prop = zend_std_get_static_property(ce, name, BP_VAR_IS);
 	EG(fake_scope) = old_scope;
 
-	if (prop) {
+	if (prop && !Z_ISUNDEF_P(prop)) {
 		RETURN_COPY_DEREF(prop);
 	}
 
 	if (def_value) {
 		RETURN_COPY(def_value);
 	}
 
-	zend_throw_exception_ex(reflection_exception_ptr, 0,
-		"Property %s::$%s does not exist", ZSTR_VAL(ce->name), ZSTR_VAL(name));
+	if (prop) {
+		zend_throw_exception_ex(reflection_exception_ptr, 0,
+			"Property %s::$%s is not initialized and no default was provided", ZSTR_VAL(ce->name), ZSTR_VAL(name));
```

Comment:
Perhaps this should say "Typed property (...)" to emphasize that this is for typed props?

Reply:
May we not just use the existing message that is already used when uninitialized accessing typed properties?

> Typed static property C::$prop must not be accessed before initialization

Reply:
Maybe, and then just as an Error rather than a ReflectionException or not?

Reply:
I've changed it to the standard wording. Although I like the custom message that I had put (as it indicates why the exception is thrown more precisely), I see the appeal of using the standard wording.